components/webapp-mgt/org.wso2.carbon.webapp.mgt/src/main/java/org/wso2/carbon/webapp/mgt/AbstractWebappDeployer.java

```diff
@@ -491,4 +500,27 @@ protected void handleRedeployment(File file) throws DeploymentException {
         deploy(data);
     }
 
+    /**
+     * Reads and returns the webapp stopped status from the registry
+     * @param webApplication WebApplication instance
+     * @return
+     */
+    private boolean isWebappStopped(WebApplication webApplication) {
```

Comment:
Here we access registry for each webapp deployed. See if we can get all webapp's status at once in a List. Idea is to reduce the number of db calls